skrub/_joiner.py

```diff
@@ -12,20 +11,23 @@
 from sklearn.utils.validation import check_is_fitted
 
 from skrub import _join_utils, _matching, _utils
-from skrub._dataframe._namespace import is_pandas, is_polars
+from skrub import _selectors as s
+from skrub._check_input import CheckInputDataFrame
+from skrub._dataframe import _common as ns
 from skrub._datetime_encoder import DatetimeEncoder
+from skrub._to_str import ToStr
 
-from . import _selectors as s
 from ._wrap_transformer import wrap_transformer
 
 
-def _as_str(column):
-    return column.fillna("").astype(str)
+def _as_str(col):
+    return ToStr().fit_transform(col)
 
 
 DEFAULT_STRING_ENCODER = make_pipeline(
     FunctionTransformer(_as_str),
     HashingVectorizer(analyzer="char_wb", ngram_range=(2, 4)),
```

Comment:
here to deal with missing values you will need to pass an appropriate preprocessor to the HashingVectorizer, as is done [here](https://github.com/skrub-data/skrub/blob/da849a5c5e7846715d6ce0c82e1f6a5fcf7ed0c5/skrub/_gap_encoder.py#L228) for the GapEncoder. you can move the private `_preprocess_text` from `_gap_encoder` to `_utils` and rename it `preprocess_text`

Reply:
sorry actually that won't be enough because the decoding will fail before we reach the preprocessor. so instead we can have `make_pipeline(FunctionTransformer(partial(sbd.fill_null, value="")), ToStr(), HashingVectorizer())`

Reply:
Thanks !